pkg/util/log/log.go

```diff
@@ -309,8 +309,12 @@ func logWithError(logLevel seelog.LogLevel, bufferFunc func(), scrubAndLogFunc f
 
 	l.l.Lock()
 
-	if l.inner == nil && !fallbackStderr {
-		addLogToBuffer(bufferFunc)
+	isInnerNil := l.inner == nil
+
+	if isInnerNil {
```

Comment:
Wouldn't it be better to release the lock when entering this branch to slightly release contention?